crates/polars-io/src/parquet/read/async_impl.rs

```diff
@@ -21,7 +21,7 @@ use crate::parquet::metadata::FileMetadataRef;
 use crate::pl_async::get_runtime;
 use crate::predicates::PhysicalIoExpr;
 
-type DownloadedRowGroup = Vec<(u64, Bytes)>;
+type DownloadedRowGroup = PlHashMap<u64, Bytes>;
```

Comment:
The Vec here was being immediately turned into a hashmap, so we just keep it as a hashmap upfront instead.